### PR TITLE
Support using AAD auth for Event Hubs Storage account

### LINKED ISSUE #1566: Azure EventHubs PubSub should use AAD for checkpointing storage
Currently EventHubs PubSub has support for AAD authentication, but still requires the storage account connection used for lease checkpointing to use a connection string instead of AAD auth.

This should be updated to also support AAD auth.

### PR DESCRIPTION
**Pull request by berndverst**
Signed-off-by: Bernd Verst <4535280+berndverst@users.noreply.github.com>

# Description

Ensures the Event Hubs binding can connect to the Storage Account for leaser / checkpointer using AAD. This requires at least Storage Blob Contributor role at the Storage Account level.

We are reusing the blob storage token from AAD credential auth logic that is well-tested and used elsewhere.

Fixes #1566

### CODE REVIEW

**Comment by berndverst** [comment]
FYI @mukundansundar

File: pubsub/azure/eventhubs/eventhubs.go [hunk 1]
```diff
@@ -140,20 +140,22 @@ type AzureEventHubs struct {
 	managementSettings azauth.EnvironmentSettings
 	cgClient           *mgmt.ConsumerGroupsClient
 	tokenProvider      *aad.TokenProvider
+	storageCredential  azblob.Credential
+	azureEnvironment   *azure.Environment
```
Comment (mukundansundar) [inline pubsub/azure/eventhubs/eventhubs.go:144]: seems this variable is not used ....

Comment (berndverst) [reply pubsub/azure/eventhubs/eventhubs.go:144]: one sec :)

Comment (berndverst) [reply pubsub/azure/eventhubs/eventhubs.go:144]: Done!

Comment (mukundansundar) [reply pubsub/azure/eventhubs/eventhubs.go:144]: still have the variable.

Comment (berndverst) [reply pubsub/azure/eventhubs/eventhubs.go:144]: Pushed the wrong branch. Try now.

**Review by surenderssm — COMMENTED** [review]
Looks good to me !!

Out of context, I do not see notion of balancing strategy while listening to an event hub. I remember using options like balance and greedy in other SDKs.

**Review by daixiang0 — CHANGES_REQUESTED** [review]
Could you add tests to cover changes?

**Comment by berndverst** [comment]
> Could you add tests to cover changes?

This PR is actually a bugfix adding missing behavior to the component. Therefore I would appreciate it if you can approve this @daixiang0.

@mukundansundar tested the component locally previously with AAD authentication but did not update the integration test. He will be creating certification tests to make this component stable. At that point we will be adding the E2E tests.

The alternative IMO is to revert the existing PRs - but that is unnecessary.

**Review by daixiang0 — APPROVED** [review]

**Pull request merged**